File: app/src/problems/python/07.py
import math
primes: [int] = [2]
num = 3
while len(primes) != 10001:
    primes_index: int = 0
    is_prime = True
    while primes[primes_index] <= math.sqrt(num):
        if num % primes[primes_index] == 0:
            is_prime = False
        primes_index +=1
    if is_prime:
        primes.append(num)
    num+=2
print(primes[-1])

File: app/src/problems/rust/07.rs
fn main() {
    let mut primes = vec![2, 3];
    let mut num = 2;
    let mut last_prime = 3;
    while num != 10001 {
        let mut i: i32 = last_prime;
        loop {
            let mut is_prime = true;
            i += 2;
            for prime in &primes {
                if (*prime as f32) > (i as f32).powf(0.5) {
                    break;
                }
                if i % prime == 0 {
                    is_prime = false;
                }
                
            }
             if is_prime {
                primes.push(i);
                last_prime = i;
                num +=1;
                break;
            }
        }
    }
    println!("{}", primes[10000]);
}
